versionedRouter.js

```diff
@@ -374,6 +374,7 @@ if (startDestTransformer) {
         { processSessions }
       );
 
+      let ctxStatusCode = 200;
```

Comment:
Backend config is also queried while performing [tracking plan validations](https://github.com/rudderlabs/rudder-transformer/blob/a2d565f082c00058a25f841b91a0d3b27526076f/versionedRouter.js#L188), shouldn't we perform similar adaptations there as well? [This call](https://github.com/rudderlabs/rudder-transformer/blob/a2d565f082c00058a25f841b91a0d3b27526076f/versionedRouter.js#L201) might fail too, right?

Reply:
tracking plan validations are soft validations. If any error is thrown other than schema validation error, event will be forwarded. There would be no dropping of events.

Reply:
Can you please further explain the above statement? What error code will rudder-server receive in case a schema validation fails due to unavailability of the control plane? We wouldn't want to drop events due to such an event, correct?

Reply:
That's correct. if control-plane is not available, response code would be 200.

Reply:
so transformer treats cp errors like the tracking plan validation is fine? Doesn't sound like the right thing to do, does it? What is the rationale behind it?

Reply:
> That's correct. if control-plane is not available, response code would be 200.

Shouldn't this be 500 so that request will be retried?

Reply:
@koladilip to provide you with some context, we cannot block the processor's pipeline. In processor we do not support retries, we need to process all events of a single batch at once.
Now, in the special case of CP unavailability we have decided that we should retry indefinitely (block the pipeline) until CP is up again, so that we don't have to mark jobs as failed, and have the customer asking us to do an event replay.

Reply:
Discussed internally. Moving to hard limitation.
@atzoum Implemented the same error handling for tracking plan api as well.